packages/core/strapi/lib/types/utils/guard.d.ts

```diff
@@ -11,4 +13,15 @@
  * type X = Never<never, string>
  * // string
  */
-export type Never<TValue, TDefault = unknown> = [TValue] extends [never] ? TDefault : TValue;
+export type Never<TValue, TFallback = unknown> = OfTypes<[never], TValue, TFallback>;
+
+export type OfTypes<TTypes extends unknown[], TValue, TFallback = unknown> = TTypes extends [
+  infer THead extends unknown,
+  ...infer TTail extends unknown[]
+]
+  ? Utils.Expression.If<
+      Utils.Expression.StrictEqual<TValue, THead>,
+      TFallback,
+      Utils.Expression.If<Utils.Array.IsNotEmpty<TTail>, OfTypes<TTail, TValue, TFallback>, TValue>
+    >
+  : never;
```

Comment:
Adding a new guard to test any type. Was very useful when debugging where the issue was coming from.